### PR TITLE
Marketplace OPSRC&CSC with publisher&displayName

### PR DESCRIPTION
**Pull request by emmajiafan**
Auto [OCP-21728 check the publisher,display,labels of opsrc&csc] @kevinrizza @ecordell  Could you help review it? Thanks very much! The test case covers OCP-21728 for marketplace. cc: @bandrade @scolange @cuipinghuo @dongboyan77 @zihantang-rh @jianzhangbjz @chengzhang1016

The logs as below:

passed: (1m33s) 2019-09-02T16:22:38 "[Feature:Marketplace] Marketplace resources with labels provider displayName [ocp-21728]create opsrc with labels [Suite:openshift/conformance/parallel]"
Timeline:

Sep 02 16:21:15.028 I ns/openshift-marketplace deployment/opsrctestlabel Scaled up replica set opsrctestlabel-76fcd58f7d to 1
Sep 02 16:21:15.028 I ns/openshift-marketplace replicaset/opsrctestlabel-76fcd58f7d Created pod: opsrctestlabel-76fcd58f7d-nbqjc
Sep 02 16:21:15.028 I ns/openshift-marketplace pod/opsrctestlabel-76fcd58f7d-nbqjc Successfully assigned openshift-marketplace/opsrctestlabel-76fcd58f7d-nbqjc to ip-10-0-154-110.ap-northeast-2.compute.internal
Sep 02 16:21:15.037 I ns/openshift-marketplace pod/opsrctestlabel-76fcd58f7d-nbqjc node/ created
Sep 02 16:21:16.559 I ns/openshift-marketplace pod/opsrctestlabel-76fcd58f7d-nbqjc Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:1ac6e4de4a360f1d33e141aa06f968de71cb4dfbe88711bcf35995f12a06c654" already present on machine
...
Sep 02 16:22:37.889 I ns/openshift-marketplace pod/opsrctestlabel-76fcd58f7d-nbqjc Stopping container opsrctestlabel
Sep 02 16:22:38.735 W ns/openshift-marketplace pod/csctestlabel-6f7dbdcb9d-9jxbr node/ip-10-0-154-110.ap-northeast-2.compute.internal graceful deletion within 30s
Sep 02 16:22:38.746 I ns/openshift-marketplace pod/csctestlabel-6f7dbdcb9d-9jxbr Stopping container csctestlabel

Writing JUnit report to junit_e2e_20190902-162238.xml

1 pass, 0 skip (1m33s)

### CODE REVIEW

File: test/extended/marketplace/marketplace_lables.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+var _ = g.Describe("[Feature:Marketplace] Marketplace resources with labels provider displayName", func() {
+
+	defer g.GinkgoRecover()
+
+	var (
+		oc            = exutil.NewCLI("marketplace", exutil.KubeConfigPath())
+		allNs         = "openshift-operators"
+		marketplaceNs = "openshift-marketplace"
+
+		//buildPruningBaseDir = exutil.FixturePath("testdata", "marketplace")
```
Comment (kevinrizza) [inline test/extended/marketplace/marketplace_lables.go]: Should this commented out code be removed?

File: test/extended/marketplace/marketplace_lables.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			err := clearResources(oc, source[0], source[1], source[2])
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+	})
+
+	//OCP-21728 check the publisher,display,labels of opsrc&csc
+	g.It("[ocp-21728]create opsrc with labels", func() {
+
+		//create one opsrc with label
+		//opsrcYaml := exutil.FixturePath("testdata", "marketplace", "opsrc", "02-opsrc.yaml")
```
Comment (kevinrizza) [inline test/extended/marketplace/marketplace_lables.go]: Same question about commented code

File: test/extended/marketplace/marketplace_lables.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		}
+
+		//get the packagelist of opsrctestlabel
+		packageListOpsrc1, _ := getResourceByPath(oc, "operatorsource", "opsrctestlabel", "-o=jsonpath={.status.packages}", marketplaceNs)
+		packageList := strings.Split(packageListOpsrc1, ",")
+
+		//get the packagelist with label of opsrctestlabel
+		packageListOpsrc2, _ := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifests", "-lopsrc-provider=optestlabel", "-o=name", "-n", marketplaceNs).Output()
+		for _, packages := range packageList {
+			o.Expect(packageListOpsrc2).Should(o.ContainSubstring(packages))
```
Comment (kevinrizza) [inline test/extended/marketplace/marketplace_lables.go]: I'm confused by this last part. I thought the purpose of this test was to check label propogation on marketplace child resources? Why are we checking the packagemanifest here? Or am I misunderstanding the purpose of this test?

File: test/extended/testdata/marketplace/csc/02-csc.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    csDisplayName: "${DISPLAYNAME}"
+    csPublisher: "${PUBLISHER}"
+
+parameters:
+- name: NAME
+- name: NAMESPACE
+- name: MARKETPLACE
+- name: PACKAGES
+- name: DISPLAYNAME
+- name: PUBLISHER
```
Comment (kevinrizza) [inline test/extended/testdata/marketplace/csc/02-csc.yaml]: This needs a newline at the end of the file.

File: test/extended/testdata/marketplace/opsrc/02-opsrc.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    displayName: "${DISPLAYNAME}"
+    publisher: "${PUBLISHER}"
+
+parameters:
+- name: NAME
+- name: NAMESPACE
+- name: MARKETPLACE
+- name: LABEL
+- name: DISPLAYNAME
+- name: PUBLISHER
```
Comment (kevinrizza) [inline test/extended/testdata/marketplace/opsrc/02-opsrc.yaml]: This also needs a newline at the end of the file.

File: test/extended/marketplace/marketplace_lables.go (hunk 1 above)
Comment (emmajiafan) [reply test/extended/marketplace/marketplace_lables.go]: will remove

File: test/extended/marketplace/marketplace_lables.go (hunk 2 above)
Comment (emmajiafan) [reply test/extended/marketplace/marketplace_lables.go]: will remove

File: test/extended/marketplace/marketplace_lables.go (hunk 3 above)
Comment (emmajiafan) [reply test/extended/marketplace/marketplace_lables.go]: This test will check label propogation on marketplace child resources and the packagemanifests from the child "catalogsource" ,The packcagemanifests will alsoo attache the label and they can be filtered by the label.

File: test/extended/testdata/marketplace/csc/02-csc.yaml (hunk 4 above)
Comment (emmajiafan) [reply test/extended/testdata/marketplace/csc/02-csc.yaml]: will add

File: test/extended/testdata/marketplace/opsrc/02-opsrc.yaml (hunk 5 above)
Comment (emmajiafan) [reply test/extended/testdata/marketplace/opsrc/02-opsrc.yaml]: will add

File: test/extended/marketplace/marketplace_utils.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+			return false, nil
+		}
+
+		e2e.Failf("Can't get resource:%s", name)
+		return false, err
+	}
+	return true, nil
+}
+
+func isResourceItemsEmpty(resourceList map[string]interface{}) bool {
```
Comment (jianzhangbjz) [inline test/extended/marketplace/marketplace_utils.go]: @emmajiafan  Do we still need to use this function? If not, please remove it and other functions which we don't need.

File: test/extended/marketplace/marketplace_lables.go [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+	//OCP-21728 check the publisher,display,labels of opsrc&csc
+	g.It("[ocp-21728]create opsrc with labels", func() {
+
+		//create one opsrc with label
+		opsrcYaml, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", opsrcYamltem, "-p", "NAME=opsrctestlabel", "NAMESPACE=marketplace_e2e", "LABEL=optestlabel", "DISPLAYNAME=optestlabel", "PUBLISHER=optestlabel", fmt.Sprintf("MARKETPLACE=%s", marketplaceNs)).OutputToFile("config.json")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = createResources(oc, opsrcYaml)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(30 * time.Second)
```
[the remark this answers is not in the document]
Comment (emmajiafan) [reply test/extended/marketplace/marketplace_lables.go]: Update to use wait.Poll

File: test/extended/marketplace/marketplace_utils.go (hunk 6 above)
Comment (emmajiafan) [reply test/extended/marketplace/marketplace_utils.go]: already remove

File: test/extended/marketplace/marketplace_lables.go [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+		//check the displayname,provider,labels of opsrc & catalogsource
+		for _, source := range opsrcResourceList {
+			msg, _ := getResourceByPath(oc, source[0], source[1], source[2], source[3])
+			o.Expect(msg).Should(o.ContainSubstring("optestlabel"))
+		}
+		//create one csc with provider&display&labels
+		cscYaml, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", cscYamltem, "-p", "NAME=csctestlabel", fmt.Sprintf("NAMESPACE=%s", allNs), fmt.Sprintf("MARKETPLACE=%s", marketplaceNs), "PACKAGES=descheduler-test", "DISPLAYNAME=csctestlabel", "PUBLISHER=csctestlabel").OutputToFile("config.json")
+		err = createResources(oc, cscYaml)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(15 * time.Second)
```
[the remark this answers is not in the document]
Comment (emmajiafan) [reply test/extended/marketplace/marketplace_lables.go]: Update to use wait.Poll

**Pull request merged**